autovalet_husky/src/autovalet_husky/find_parking_spot.py

```diff
@@ -69,8 +70,12 @@ def pub_two_goals(self):
         self.goal1 = self.generate_parking_goal(self.tag_tf, pos1, rot1)
         while not parking:
             rospy.sleep(0.5)
+            # keep moving w/ goal gen until goal1 is in costmap
+            if self.dist_to_goal(self.goal1) > 4.0:
+                continue
             # keep publishing the first goal until the robot is close enough
-            if self.dist_to_first_goal(self.goal1) > 0.5:
+            self.first_goal_close = True
+            if self.dist_to_goal(self.goal1) > 0.5:
```

Comment:
Technically these should be class members. self.first_goal_tol = 0.5 etc. Upto you if you wanna fix them!

Reply:
fixed